<a href="https://colab.research.google.com/github/lovishkanther11/Deep-Learning/blob/master/Keras_metrics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Usage of metrics

A metric is a function that is used to judge the performance of your model. Metric functions are to be supplied in the metrics parameter when a model is compiled.

```
model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=['mae', 'acc'])
from keras import metrics

model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=[metrics.mae, metrics.categorical_accuracy])
```

A metric function is similar to a loss function, except that the results from evaluating a metric are not used when training the model. You may use any of the loss functions as a metric function.

You can either pass the name of an existing metric, or pass a Theano/TensorFlow symbolic function (see Custom metrics).

##Arguments
1. y_true: True labels. Theano/TensorFlow tensor.
2. y_pred: Predictions. Theano/TensorFlow tensor of the same shape as y_true.

##Returns

1. Single tensor value representing the mean of the output array across all datapoints.

#Accuracy
keras.metrics.accuracy(y_true, y_pred)

In [1]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, y_train), (X_test, y_test) = mnist.load_data()

# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
y_train = to_categorical(y_train, NUM_CLASSES)
y_test = to_categorical(y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='elu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='elu'))
model.add(Dense(10, activation='softmax'))

# Compile model using accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Using TensorFlow backend.


11493376/11490434 [==============================] - 0s 0us/step





Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where



Train on 60000 samples, validate on 10000 samples
Epoch 1/10





60000/60000 [==============================] - 11s 191us/step - loss: 0.2901 - acc: 0.9114 - val_loss: 0.1740 - val_acc: 0.9436
Epoch 2/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.1222 - acc: 0.9626 - val_loss: 0.1131 - val_acc: 0.9649
Epoch 3/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0836 - acc: 0.9739 - val_loss: 0.0945 - val_acc: 0.9718
Epoch 4/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0634 - acc: 0.9800 - val_loss: 0.0888 - val_acc: 0.9723
Epoch 5/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0510 - acc: 0.9837 - val_loss: 0.0810 - val_acc: 0.9750
Epoch 6/10
60000/60000 [==============================] - 2s 32us/step - loss: 0.0414 - acc

#binary_accuracy

keras.metrics.binary_accuracy(y_true, y_pred, threshold=0.5)

In [2]:
# Compile model using binary_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['binary_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 35us/step - loss: 0.0188 - binary_accuracy: 0.9988 - val_loss: 0.0881 - val_binary_accuracy: 0.9961
Epoch 2/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0163 - binary_accuracy: 0.9990 - val_loss: 0.1066 - val_binary_accuracy: 0.9959
Epoch 3/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0140 - binary_accuracy: 0.9991 - val_loss: 0.1055 - val_binary_accuracy: 0.9961
Epoch 4/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0118 - binary_accuracy: 0.9992 - val_loss: 0.1032 - val_binary_accuracy: 0.9962
Epoch 5/10
60000/60000 [==============================] - 2s 33us/step - loss: 0.0111 - binary_accuracy: 0.9993 - val_loss: 0.1100 - val_binary_accuracy: 0.9961
Epoch 6/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0110 - binary_accuracy: 0.9993 - val_loss: 0.1122 - val_binary_a

#categorical_accuracy

keras.metrics.categorical_accuracy(y_true, y_pred)

In [3]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 34us/step - loss: 0.0079 - categorical_accuracy: 0.9977 - val_loss: 0.1220 - val_categorical_accuracy: 0.9811
Epoch 2/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0066 - categorical_accuracy: 0.9981 - val_loss: 0.1182 - val_categorical_accuracy: 0.9813
Epoch 3/10
60000/60000 [==============================] - 2s 32us/step - loss: 0.0059 - categorical_accuracy: 0.9983 - val_loss: 0.1320 - val_categorical_accuracy: 0.9806
Epoch 4/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0056 - categorical_accuracy: 0.9984 - val_loss: 0.1303 - val_categorical_accuracy: 0.9817
Epoch 5/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0062 - categorical_accuracy: 0.9984 - val_loss: 0.1326 - val_categorical_accuracy: 0.9806
Epoch 6/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0062 - categor

#top_k_categorical_accuracy

keras.metrics.top_k_categorical_accuracy(y_true, y_pred, k=5)

In [4]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 36us/step - loss: 0.0059 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1421 - val_top_k_categorical_accuracy: 0.9999
Epoch 2/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.0054 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1343 - val_top_k_categorical_accuracy: 0.9999
Epoch 3/10
60000/60000 [==============================] - 2s 32us/step - loss: 0.0046 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1569 - val_top_k_categorical_accuracy: 1.0000
Epoch 4/10
60000/60000 [==============================] - 2s 33us/step - loss: 0.0048 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1506 - val_top_k_categorical_accuracy: 0.9999
Epoch 5/10
60000/60000 [==============================] - 2s 32us/step - loss: 0.0031 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1663 - val_top_k_categorical_accuracy: 0.9997
Epoch 6/10
60000/60000 [===========

#sparse_categorical_accuracy

keras.metrics.sparse_categorical_accuracy(y_true, y_pred)

In [5]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7395 - sparse_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.7392 - sparse_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.7389 - sparse_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.7385 - sparse_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.7381 - sparse_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.7377 - sparse_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.7374 - sparse_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.7370 - sparse_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.7366 - sparse_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.7362 - sparse_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.7358 - sparse_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.7355 - sparse_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.7351 - sparse_categorical_accuracy: 0.5000
Epoch 14/400
 - 0s - loss: 0.7347 - sparse_categorical_accuracy: 0.5000
E

#sparse_top_k_categorical_accuracy

keras.metrics.sparse_top_k_categorical_accuracy(y_true, y_pred, k=5)

In [6]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_top_k_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7828 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.7821 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.7814 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.7808 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.7801 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.7794 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.7788 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.7781 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.7775 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.7768 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.7762 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.7756 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.7749 - sparse_top_k_categorical_accuracy: 0.

#cosine_proximity

keras.metrics.cosine_proximity(y_true, y_pred, axis=-1)

In [7]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['cosine_proximity'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.6855 - cosine_proximity: -5.0000e-01
Epoch 2/400
 - 0s - loss: 0.6854 - cosine_proximity: -5.0000e-01
Epoch 3/400
 - 0s - loss: 0.6852 - cosine_proximity: -5.0000e-01
Epoch 4/400
 - 0s - loss: 0.6851 - cosine_proximity: -5.0000e-01
Epoch 5/400
 - 0s - loss: 0.6850 - cosine_proximity: -5.0000e-01
Epoch 6/400
 - 0s - loss: 0.6849 - cosine_proximity: -5.0000e-01
Epoch 7/400
 - 0s - loss: 0.6847 - cosine_proximity: -5.0000e-01
Epoch 8/400
 - 0s - loss: 0.6846 - cosine_proximity: -5.0000e-01
Epoch 9/400
 - 0s - loss: 0.6845 - cosine_proximity: -5.0000e-01
Epoch 10/400
 - 0s - loss: 0.6843 - cosine_proximity: -5.0000e-01
Epoch 11/400
 - 0s - loss: 0.6842 - cosine_proximity: -5.0000e-01
Epoch 12/400
 - 0s - loss: 0.6841 - cosine_proximity: -5.0000e-01
Epoch 13/400
 - 0s - loss: 0.6840 - cosine_proximity: -5.0000e-01
Epoch 14/400
 - 0s - loss: 0.6838 - cosine_proximity: -5.0000e-01
Epoch 15/400
 - 0s - loss: 0.6837 - cosine_proximity: -5.0000e-01
Epoch 16/400
 - 0s 

#Thank you for completing the notebook

*ACCURACY SCORES*

**Accuracy-**

Test loss: 0.10376196658944027


Test accuracy: 0.9757


**binary_accuracy-**

Test loss: 0.15646206490577766


Test accuracy: 0.9955399974822998


**categorical_accuracy-**

Test loss: 0.14195446051103405


Test accuracy: 0.9816


**top_k_categorical_accuracy-**

Test loss: 0.15224556388646485


Test accuracy: 0.9999


**sparse_categorical_accuracy-**

loss: 0.4375 


sparse_categorical_accuracy: 0.5000


**sparse_top_k_categorical_accuracy-**

loss: 0.7547 


sparse_top_k_categorical_accuracy: 0.5000


**cosine_proximity-**

loss: 0.7574 


cosine_proximity: -5.0000e-01

The best metric among all is **top_k_categorical_accuracy** which gives almost an accuracy of 1.

In [8]:
from keras.datasets import cifar10
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar10.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 10)
Y_test =  to_categorical(Y_test, 10)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(10))
model.add(Activation('softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

170500096/170498071 [==============================] - 2s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 3s 67us/step - loss: 2.2473 - top_k_categorical_accuracy: 0.7505 - val_loss: 1.7536 - val_top_k_categorical_accuracy: 0.8568
Epoch 2/10
50000/50000 [==============================] - 3s 54us/step - loss: 1.8082 - top_k_categorical_accuracy: 0.8509 - val_loss: 1.6659 - val_top_k_categorical_accuracy: 0.8857
Epoch 3/10
50000/50000 [==============================] - 3s 54us/step - loss: 1.7041 - top_k_categorical_accuracy: 0.8728 - val_loss: 1.8074 - val_top_k_categorical_accuracy: 0.8317
Epoch 4/10
50000/50000 [==============================] - 3s 56us/step - loss: 1.6237 - top_k_categorical_accuracy: 0.8880 - val_loss: 1.6432 - val_top_k_categorical_accuracy: 0.8910
Epoch 5/10
50000/50000 [==============================] - 3s 55us/step - loss: 1.5699 - top_k_categorical_accuracy: 0.8992 - val_loss: 1.6008 - val_top

**CIFAR 10**

Test loss: 1.5381210020065308


Test accuracy: 0.9088

In [0]:
from keras.datasets import cifar100
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar100.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 100)
Y_test =  to_categorical(Y_test, 100)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(100))
model.add(Activation('softmax'))


from keras import optimizers
adamax=keras.optimizers.Adamax(lr=0.002, beta_1=0.9, beta_2=0.999)

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

169009152/169001437 [==============================] - 4s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 3s 64us/step - loss: 4.2952 - top_k_categorical_accuracy: 0.2036 - val_loss: 3.9600 - val_top_k_categorical_accuracy: 0.2928
Epoch 2/10
50000/50000 [==============================] - 3s 52us/step - loss: 3.7820 - top_k_categorical_accuracy: 0.3419 - val_loss: 3.7000 - val_top_k_categorical_accuracy: 0.3734
Epoch 3/10
50000/50000 [==============================] - 3s 51us/step - loss: 3.6018 - top_k_categorical_accuracy: 0.3965 - val_loss: 3.6282 - val_top_k_categorical_accuracy: 0.3977
Epoch 4/10
50000/50000 [==============================] - 3s 51us/step - loss: 3.4651 - top_k_categorical_accuracy: 0.4316 - val_loss: 3.6416 - val_top_k_categorical_accuracy: 0.3966
Epoch 5/10
50000/50000 [==============================] - 3s 50us/step - loss: 3.3696 - top_k_categorical_accuracy: 0.4536 - val_loss: 3.4982 - val_top

**CIFAR 100**

Test loss: 3.393477967834473


Test accuracy: 0.483

In [0]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import fashion_mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, Y_train), (X_test, Y_test) = fashion_mnist.load_data()



# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
Y_train = to_categorical(Y_train, NUM_CLASSES)
Y_test = to_categorical(Y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='relu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='relu'))
model.add(Dense(10, activation='softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 38us/step - loss: 0.5428 - top_k_categorical_accuracy: 0.9932 - val_loss: 0.6340 - val_top_k_categorical_accuracy: 0.9960
Epoch 2/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.3741 - top_k_categorical_accuracy: 0.9978 - val_loss: 0.3925 - val_top_k_categorical_accuracy: 0.9978
Epoch 3/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.3345 - top_k_categorical_accuracy: 0.9984 - val_loss: 0.3887 - val_top_k_categorical_accuracy: 0.9981
Epoch 4/10
60000/60000 [==============================] - 2s 31us/step - loss: 0.3086 - top_k_categorical_accuracy: 0.9986 - val_loss: 0.3435 - val_top_k_categorical_accuracy: 0.9975
Epoch 5/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.2895 - top_k_categorical_accuracy: 0.9990 - val_loss: 0.3534 - val_top_k_categorical_accuracy: 0.9980
Epoch 6/10
60000/60000 [===========

**FASHION_MNIST**

Test loss: 0.34572943717837334


Test accuracy: 0.9981